Assets/Engine/Chat/ChatWindow.cs

```diff
@@ -218,15 +221,27 @@ public bool PlayerIsTyping()
             return (EventSystem.current.currentSelectedGameObject != null);
         }
 
-        public void FinishTyping()
+        public void FocusInputField()
         {
-            SendMessage();
-            EventSystem.current.SetSelectedGameObject(null);
+            inputField.Select();
         }
 
-        public void FocusInputField()
+        public void onSelect()
         {
-            inputField.Select();
+            this.isSelected = true;
+        }
+
+        public void onDeselect()
+        {
+            this.isSelected = false;
+        }
+
+        public void Update()
+        {
+            if ((Input.GetKeyUp(KeyCode.Return) || Input.GetKeyUp(KeyCode.KeypadEnter) && this.isSelected)
```

Comment:
I'm getting a compile-time error here. There are more open brackets than close brackets.

Reply:
Whoops should have tested that, it's fixed now.